src/annotator/test/notebook-test.js

```diff
@@ -34,14 +45,24 @@ describe('Notebook', () => {
     it('creates the container', () => {
       assert.isFalse(container.hasChildNodes());
       const notebook = createNotebook();
-      assert.calledOnce(render);
-      assert.isNotNull(notebook._outerContainer.shadowRoot);
+      const shadowRoot = notebook._outerContainer.shadowRoot;
+      assert.isNotNull(shadowRoot);
+      assert.isNotNull(shadowRoot.querySelector('#notebook-modal'));
     });
 
     it('removes the container', () => {
       const notebook = createNotebook();
       notebook.destroy();
       assert.isFalse(container.hasChildNodes());
     });
+
+    /*
+    it('calls the clean up function of the NotebookModal when the container is removed', () => {
```

Comment:
I tried unsuccessfully to add a test case to check that the useEffect's cleanup function is executed.

Reply:
You need to wrap the `createNotebook` and `destroy` calls in separate `act` calls. The first `act` block will ensure that the cleanup function is registered and the second one will actually run it. Note that using a single `act` block didn't work in my tests - I guess that is because unmounting the component cancels the pending effect.